scribejava-apis/src/main/java/com/github/scribejava/apis/service/GeniusOAuthServiceImpl.java

```diff
@@ -0,0 +1,39 @@
+package com.github.scribejava.apis.service;
+
+import com.github.scribejava.core.builder.api.DefaultApi20;
+import com.github.scribejava.core.model.AbstractRequest;
+import com.github.scribejava.core.model.OAuth2AccessToken;
+import com.github.scribejava.core.model.OAuthConfig;
+import com.github.scribejava.core.model.OAuthConstants;
+import com.github.scribejava.core.oauth.OAuth20Service;
+
+public class GeniusOAuthServiceImpl extends OAuth20Service {
+	private static final String USER_AGENT = "user-agent";
+	private static final String AGENT_NAME = "example_name";
```

Comment:
Hi. I had no luck to "remove service as argument from Request's constructor".
But I have refactored async configurations a bit
https://github.com/scribejava/scribejava/commit/2751f5293b75a5f62e2265802508fb721e8d72d7
And have added userAgent config option
https://github.com/scribejava/scribejava/commit/c6d55483e117dfb8a5f8cc2353c5dec4d3c1cb00

Have updated your commits to this one
https://github.com/scribejava/scribejava/commit/338406dc2c716e15f96aae43614628f8b285502f
and have merged it to the master. Have checked it with credentials provided by you. It's working ;-)

Here we have to add userAgent in GeniusExample
https://github.com/scribejava/scribejava/commit/338406dc2c716e15f96aae43614628f8b285502f#diff-eb092bcea18fab8bac3b3d33a65e9f0eR29

ps.In Async version user can user com.ning.http.client.AsyncHttpClientConfig::setUserAgent as an option as well.